create-publication-from-scopus/src/main/java/no/sikt/nva/scopus/ScopusHandler.java

```diff
@@ -32,17 +47,50 @@ public ScopusHandler(S3Client s3Client) {
     }
 
     @Override
-    public String handleRequest(S3Event event, Context context) {
-        return attempt(() -> readFile(event))
-            .map(this::parseXmlFile)
-            .map(this::getDoi)
-            .orElseThrow(fail -> logErrorAndThrowException(fail.getException()));
+    public CreatePublicationRequest handleRequest(S3Event event, Context context) {
+        CreatePublicationRequest request = attempt(() -> readFile(event))
+                .map(this::parseXmlFile)
+                .map(this::extractMetadata)
+                .orElseThrow(fail -> logErrorAndThrowException(fail.getException()));
+        return request;
+    }
+
+    private CreatePublicationRequest extractMetadata(DocTp docTp) {
+        CreatePublicationRequest request = new CreatePublicationRequest();
+        EntityDescription entityDescription = new EntityDescription();
+        request.setEntityDescription(entityDescription);
+        Reference reference = new Reference();
+        entityDescription.setReference(reference);
+        reference.setPublicationContext(getPublicationContext(docTp));
+        return request;
     }
 
-    private String getDoi(DocTp docTp) {
-        var doi= docTp.getMeta().getDoi();
-        logger.info("The publication doi:"+doi);
-        return doi;
+    private PublicationContext getPublicationContext(DocTp docTp) {
+        PublicationContext publicationContext = null;
+        ScopusSourceType scopusSourceType = ScopusSourceType.valueOfCode(docTp.getMeta().getSrctype());
+        if (Objects.requireNonNull(scopusSourceType) == JOURNAL) {
+            StringBuilder publisherName = new StringBuilder();
+            SourceTp sourceTp = docTp.getItem().getItem().getBibrecord().getHead().getSource();
+            sourceTp.getSourcetitle().getContent().forEach(publisherName::append);
+            try {
+                String electronicIssn = EMPTY_STRING;
+                String printedIssn = EMPTY_STRING;
+                for (IssnTp issnTp : sourceTp.getIssn()) {
+                    String issn = issnTp.getContent();
+                    issn = issn.substring(0, 4) + DASH + issn.substring(4);
+                    if (ISSN_TYPE_ELECTRONIC.equals(issnTp.getType())) {
+                        electronicIssn = issn;
+                    } else if (ISSN_TYPE_PRINT.equals(issnTp.getType())) {
+                        printedIssn = issn;
+                    }
+                }
+                publicationContext = new UnconfirmedJournal(publisherName.toString(), printedIssn,
+                        electronicIssn);
+            } catch (InvalidIssnException e) {
+                logger.error(e.getMessage());
+            }
```

Comment:
I'd probably do something like this:
`

    private UnconfirmedJournal getNullContextLoggingError(Failure<UnconfirmedJournal> fail) {
        logger.error(fail.getException().getMessage());
        return null;
    }

    private UnconfirmedJournal extractContext(DocTp docTp) throws InvalidIssnException {
        var source = getSource(docTp);
        var issnTpList = source.getIssn();
        return new UnconfirmedJournal(extractPublisherName(source), findPrintIssn(issnTpList),
                findElectronicIssn(issnTpList));
    }

    private String findElectronicIssn(List<IssnTp> issnTpList) {
        return findIssn(issnTpList, ISSN_TYPE_ELECTRONIC);
    }

    private String findPrintIssn(List<IssnTp> issnTpList) {
        return findIssn(issnTpList, ISSN_TYPE_PRINT);
    }

    private String findIssn(List<IssnTp> issnTpList, String issnType) {
        return issnTpList.stream()
            .filter(issn -> issnType.equals(issn.getType()))
            .map(IssnTp::getContent)
            .map(this::addDashToIssn)
            .findAny()
            .orElse(null);
    }

    private String addDashToIssn(String issn) {
        return issn.substring(0, 4) + DASH + issn.substring(4);
    }

    private String extractPublisherName(SourceTp sourceTp) {
        StringBuilder publisherName = new StringBuilder();
        sourceTp.getSourcetitle().getContent().forEach(publisherName::append);
        return publisherName.toString();
    }

    private SourceTp getSource(DocTp docTp) {
        return docTp.getItem().getItem().getBibrecord().getHead().getSource();
    }
`

Reply:
I like that. Ist good.